src/metrics/transport_message_metrics.py

```diff
@@ -1,33 +1,51 @@
 import logging
+from typing import TypedDict
 
 from metrics.metrics import DEPOSIT_MESSAGES, PAUSE_MESSAGES, PING_MESSAGES, UNVET_MESSAGES
 from transport.msg_providers.rabbit import MessageType
-from transport.msg_types.deposit import DepositMessage
 
 logger = logging.getLogger(__name__)
 
 
-def message_metrics_filter(msg: DepositMessage) -> bool:
-    msg_type = msg.get('type', None)
-    logger.info({'msg': 'Guardian message received.', 'value': msg, 'type': msg_type})
-
-    address, version = msg.get('guardianAddress'), msg.get('app', {}).get('version')
+def message_metrics_filter(msg: TypedDict) -> bool:
+    """
+    Processes guardian messages and updates Prometheus metrics based on the message type.
+    Returns True for valid message types to allow further processing, and False for messages
+    that should be filtered (such as PING messages).
 
-    if msg_type == MessageType.PAUSE:
-        PAUSE_MESSAGES.labels(address, msg.get('stakingModuleId', -1), version).inc()
-        return True
+    Args:
+        msg: A dictionary containing message details.
 
-    if msg_type == MessageType.DEPOSIT:
-        DEPOSIT_MESSAGES.labels(address, msg.get('stakingModuleId', -1), version).inc()
-        return True
+    Returns:
+        bool: True if the message should be processed, False otherwise.
+    """
+    msg_type = msg.get('type')
+    logger.info({'msg': 'Guardian message received.', 'value': msg, 'type': msg_type})
 
-    if msg_type == MessageType.UNVET:
-        UNVET_MESSAGES.labels(address, msg.get('stakingModuleId', -1), version).inc()
+    address = msg.get('guardianAddress')
+    version = msg.get('app', {}).get('version')
+    transport = msg.get('transport', '')
+    chain_id = msg.get('chain_id', '')
+    staking_module_id = msg.get('stakingModuleId', -1)
+
+    metrics_map = {
+        MessageType.PAUSE: PAUSE_MESSAGES,
+        MessageType.DEPOSIT: DEPOSIT_MESSAGES,
+        MessageType.UNVET: UNVET_MESSAGES,
+    }
+
+    if msg_type in metrics_map:
+        metrics_map[msg_type].labels(
+            address=address,
+            staking_module_id=staking_module_id,
+            version=version,
+            transport=transport,
+            chain_id=chain_id,
+        ).inc()
         return True
 
     if msg_type == MessageType.PING:
```

Comment:
Are we sure we have to handle the pauses separately now? Because we send them out once every 300 blocks.

Reply:
This filter is filtering out pings because we are using them only for metrics, so the ping messages won't go to the main logic of depositor/pauser/unvetter bots